### PR TITLE
Fall back to trace hint when eval_upper_bound returns int_oo

### PR DESCRIPTION
**Pull request by psiddh**
When bound_sympy cannot compute a finite upper bound for a symbolic expression (returns int_oo), fall back to eval_expr for the trace hint before returning int_oo. This allows models with unbounded dynamic shapes (e.g. Dim('batch', min=1) with no max) to pass through ConstraintBasedSymShapeEvalPass without raising.

Previously, int_oo was returned directly, causing
ConstraintBasedSymShapeEvalPass to raise RuntimeError because it requires concrete int upper bounds for memory planning. Downstream exporters had to monkey-patch eval_upper_bound to work around this.

### CODE REVIEW

**Comment by JacobSzwejbka** [comment]
Isnt it sort of unexpected behavior if it falls back to wrong shapes from user input and not following the constraints?

**Comment by psiddh** [comment]
> Isnt it sort of unexpected behavior if it falls back to wrong shapes from user input and not following the constraints?

The trace hint isn't - wrong shapes from user input. - User says Dim('batch', min=1) -->  user can't export at all and crashes  which is the current behavior. The trace hint is the concrete value seen during torch.export() — it's the best available information when constraints give int_oo. Without this fallback, to_executorch() crashes unconditionally for any Dim(min=X) without explicit max (looks like a common test-cases in  hf-exporters repo).

File: exir/sym_util.py [hunk 1]
```diff
@@ -63,7 +63,13 @@ def eval_upper_bound(maybe_symint: Union[int, torch.SymInt]) -> int:
...
         return concrete_upper
     elif int_oo is not None and upper_bound is int_oo:
-        return int_oo
+        hint = eval_expr(maybe_symint)
+        if hint is not None:
+            return hint
+        raise RuntimeError(
+            f"Cannot evaluate a finite upper bound for symbolic expression {expr} "
+            "(int_oo) and no trace hint is available."
+        )
```
Comment (digantdesai) [inline exir/sym_util.py:72]: can runtime handle this with some large number or should we block export here?

**Review by digantdesai — APPROVED** [review]

File: exir/sym_util.py (hunk 1 above)
Comment (psiddh) [reply exir/sym_util.py:72]: the return value becomes a physical memory allocation in the .pte, I think. so a large value would just cause OOM at runtime instead of a clear error at export
